Fix logs view

TW80000 opened the pull request:
It was previously a custom view that duplicated a lot of behaviour that
already exists in ScrollTableView and DAOController. In this commit we
remove the custom view and replace it with a custom DAOController. Doing
so fixes the visual bugs and removes the redundant search box.

Fixes: https://nanopay.atlassian.net/browse/CPF-794

Commit:
Fix logs view

It was previously a custom view that duplicated a lot of behaviour that
already exists in ScrollTableView and DAOController. In this commit we
remove the custom view and replace it with a custom DAOController. Doing
so fixes the visual bugs and removes the redundant search box.

kgrgreer reviewed (changes requested):
Don't merge, will provide more comments.

jlhughes reviewed (dismissed):
Well that's cool. 
Thanks

kgrgreer commented:
I added colours to LogLevels, so you can now just do ".style({ color: severity.color })" and remove the switch statement.

commit 53d7ed41e702738b84d3c8e5bb7771d640b73745
Author: Kevin Greer <kgrgreer@gmail.com>

Date:   Mon Apr 1 11:57:05 2019 -0400
    Add colours to LogLevels.

Commit:
Merge branch 'master' into fix-logs-view

TW80000 commented:
> I added colours to LogLevels, so you can now just do ".style({ color: severity.color })" and remove the switch statement.
> 
> commit [53d7ed4](https://github.com/foam-framework/foam2/commit/53d7ed41e702738b84d3c8e5bb7771d640b73745)
> Author: Kevin Greer [kgrgreer@gmail.com](mailto:kgrgreer@gmail.com)
> 
> Date: Mon Apr 1 11:57:05 2019 -0400
> Add colours to LogLevels.

Looks like we have two `LogLevel` models, and in that commit you modified the one we aren't using for `LogMessage`s. There's `foam.log.LogLevel` and `foam.nanos.logger.LogLevel`. Looks like `foam.log.LogLevel` is used in 3 files and `foam.nanos.logger.LogLevel` is used in 4 files.

~@kgrgreer Do you want me to refactor to get rid of one of the two or just add a color property to the other one?~

Will just refactor to remove the duplicate. It shouldn't take long.

kgrgreer commented:
Just use foam.log.LogLevel if we can.

On Mon, 1 Apr 2019 at 12:14, Troy Wolters <notifications@github.com> wrote:

> I added colours to LogLevels, so you can now just do ".style({ color:
> severity.color })" and remove the switch statement.
>
> commit 53d7ed4
> <https://github.com/foam-framework/foam2/commit/53d7ed41e702738b84d3c8e5bb7771d640b73745>
> Author: Kevin Greer kgrgreer@gmail.com
>
> Date: Mon Apr 1 11:57:05 2019 -0400
> Add colours to LogLevels.
>
> Looks like we have two LogLevel models, and in that commit you modified
> the one we aren't using for LogMessages. There's foam.log.LogLevel and
> foam.nanos.logger.LogLevel. Looks like foam.log.LogLevel is used in 3
> files and foam.nanos.logger.LogLevel is used in 4 files.
>
> @kgrgreer <https://github.com/kgrgreer> Do you want me to refactor to get
> rid of one of the two or just add a color property to the other one?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/foam-framework/foam2/pull/2003#issuecomment-478644884>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHHXs9tnuv7j2kv5FmmmNoMYlJY1oEagks5vcjBigaJpZM4cV__q>
> .
>

Commit:
Remove duplicate LogLevel model

Also refactored how the colours are set in the tableCellFormatter for
the `LogMessage.severity` property.

TW80000 commented:
> [...] so you can now just do ".style({ color: severity.color })" and remove the switch statement.

> Just use foam.log.LogLevel if we can.

Done in 6a13496.

kgrgreer commented on src/foam/nanos/logger/LogMessageController.js:
```diff
@@ -0,0 +1,40 @@
+  name: 'LogMessageController',
+  extends: 'foam.comics.DAOController',
+
+  documentation: 'A custom DAOController to work with log messages.',
```
Add a comment on what's custom about it and why it's required.

kgrgreer commented on src/foam/nanos/logger/LogMessageController.js:
```diff
@@ -0,0 +1,40 @@
+    }
+  ],
+
+  classes: [
```
Is this new class needed? Couldn't you just scope the CSS in the LogMessageController?

TW80000 replied:
I tried that but it doesn't work because `LogMessageController` extends `DAOController`, not `DAOControllerView`. And since it's not a view, it doesn't get CSS.

kgrgreer replied:
I see. Another reason to change comics to get rid of the view/model distinction, which doesn't add any value and just causes extra work and complexity.

TW80000 replied:
I guess since the fixed column width PR was merged we don't need this anymore anyway, so I'll remove it.

Commit:
Remove unnecessary code to limit table width

With the recent changes to tables, this is no longer necessary.

Commit:
Add detail to comment

Commit:
Merge branch 'master' into fix-logs-view

Commit:
Remove reference to deleted model

Commit:
Order logMessageDAO in NSpec rather than using custom view
